charms/kubernetes_snaps.py

```diff
@@ -510,6 +516,13 @@ def _dict_compare(d1, d2):
 def _calculate_config_difference(service: str, args, config_files):
     args_hash = _snap_common_path(service) / "args-hash.txt"
 
+    if not _enable_config_hashing():
+        log.debug("Config hashing disabled, skipping config change detection")
+        yield True
+        args_hash.unlink(missing_ok=True)
+        return
```

Comment:
if hashing isn't enabled, this method declares there are changes, and we should reconfigure the service (as before this feature existed) and it also clears the cache file if one exists.